scantext/views.py

```diff
@@ -0,0 +1,395 @@
+
+
+def license_scanview(request):
+    if request.method != "POST":
+        return render(
+            request, "scantext/license_scan_form.html", {"form": LicenseScanForm()}
+        )
+
+    form = LicenseScanForm(request.POST, request.FILES)
+    if not form.is_valid():
+        return render(
+            request, "scantext/license_scan_form.html", {"form": LicenseScanForm()}
+        )
+
+    input_text = form.cleaned_data["input_text"]
+    input_file = request.FILES.get("input_file", False)
+
+    if input_text and input_file:
+        message = "Provide text or a text file but not both."
+        messages.warning(request, message)
+        return render(
+            request,
+            "scantext/license_scan_form.html",
+            {
+                "form": LicenseScanForm(),
+            },
+        )
+
+    if not input_text and not input_file:
+        message = "Provide text or a text file to scan."
+        messages.warning(request, message)
+        return render(
+            request,
+            "scantext/license_scan_form.html",
+            {
+                "form": LicenseScanForm(),
+            },
+        )
+
+    # The flush in tempfile is required to ensure that the content is
+    # written to the disk before it's read by get_licenses function
+    # TODO: check this to handle input files
+    # https://github.com/nexB/commoncode/blob/9131627677d3ef171ddc472991a5c4d4a3431ee3/src/commoncode/fileutils.py#L99
+
+    if input_text:
+        with tempfile.NamedTemporaryFile(mode="w") as temp_file:
+            temp_file.write(input_text)
+            temp_file.flush()
+            expressions = get_licenses(location=temp_file.name)
+    elif input_file:
+        try:
+            # rework on how to handle temporary files.
+            with tempfile.NamedTemporaryFile(mode="w") as temp_file:
+                input_text = str(input_file.read(), "UTF-8")
+                temp_file.write(input_text)
+                temp_file.flush()
+                expressions = get_licenses(location=temp_file.name)
+        except UnicodeDecodeError:
+            message = "Please upload a valid text file."
+            messages.warning(request, message)
+            return render(
+                request,
+                "scantext/license_scan_form.html",
+                {
+                    "form": LicenseScanForm(),
+                },
+            )
+
+    if not expressions:
+        message = "Could not detect any license."
+        messages.warning(request, message)
+        return render(
+            request,
+            "scantext/license_scan_form.html",
+            {
+                "form": LicenseScanForm(),
+            },
+        )
+
+    # import json
+    # print(json.dumps(expressions, indent=2))
+
+    return render(
+        request,
+        "scantext/license_summary.html",
+        {
+            "text": input_text,
+            "detected_licenses": expressions,
+        },
+    )
+
+
+def get_licenses(
```

Comment:
Should all those non-view functions be moved into the match_text module?

Reply:
Yes tdruez!